File: Passenger.java
public class Passenger {
    private int passengerId;
    private String name;
    private String phone;
    private String email;

    public Passenger() {
    }

    public Passenger(int passengerId, String name, String phone, String email) {
        this.passengerId = passengerId;
        this.name = name;
        this.phone = phone;
        this.email = email;
    }

    public void displayPassenger() {
        System.out.println("\n========== Passenger ==========");
        System.out.println("Passenger ID : " + passengerId);
        System.out.println("Name         : " + name);
        System.out.println("Phone        : " + phone);
        System.out.println("Email        : " + email);
        System.out.println("===============================");
    }

    public String getName() {
        return name;
    }
}

File: Passenger.cpp


Passenger::Passenger()
    : User(0, "Unknown", "Unknown")
{
    email = "Unknown";
}

Passenger::Passenger(
    int userId,
    string name,
    string phone,
    string email)
    : User(userId, name, phone)
{
    this->email = email;
}

Passenger::~Passenger()
{
    cout << "Passenger object destroyed." << endl;
}

string Passenger::getEmail()
{
    return email;
}

void Passenger::setEmail(string email)
{
    this->email = email;
}

void Passenger::displayRole()
{
    cout << "Role : Passenger" << endl;
}

void Passenger::searchBus()
{
    cout << getName() << " is searching for a bus." << endl;
}

void Passenger::viewBookings()
{
    cout << getName() << " is viewing bookings." << endl;
}

void Passenger::cancelBooking()
{
    cout << getName() << " is cancelling a booking." << endl;
}

void Passenger::displayPassenger()
{
    cout << "\n========== PASSENGER ==========" << endl;

    displayUser();
    displayRole();

    cout << "Email : " << email << endl;

    cout << "===============================" << endl;
}

// Operator Overloading
// == compares two Passenger objects
bool Passenger::operator==(const Passenger& other)
{
    return userId == other.userId;
}
